kotlin-code-style.md

```diff
@@ -40,49 +29,43 @@ class Person(val firstName: String, val lastName: String, var age: Int) {
 }
 ```
 
-# Extensions
+## Constructors
 
-When adding extensions to external classes, make separate files for each type
-and name them _<Type>Extensions.kt_:
+As often as possible we should specify if the constructor properties are mutable
+(`var`) or read-only (`val`):
```

Comment:
Not sure if I agree with the phrasing of this. (That is, it was weird before you moved it, too.) It's not really about whether the arguments are mutable or not, it's about making them accessible properties on the class. Consider these two examples:

In this first case, the arguments `firstName` and `lastName` will not become actual properties on the class, meaning that this example won't compile (and probably also gives lint warnings about those constructor parameters not being used):

```kotlin
class Person(firstName: String, lastName: String)

fun main(args: Array<String>) {
    val person = Person("John", "Doe");

    // NOTE: Won't compile, as firstName and lastName are not accessible!
    print("${person.firstName} ${person.lastName}")    
}
```

However, this example **will** compile and run just fine:

```kotlin
class Person(firstName: String, lastName: String)

fun main(args: Array<String>) {
    val person = Person("John", "Doe");

    print("${person.firstName} ${person.lastName}")    
}
```